utils/controller/controller.go

```diff
@@ -18,6 +23,72 @@ import (
 	logutil "github.com/argoproj/argo-rollouts/utils/log"
 )
 
+// processNextWatchObj will process a single object from the watch by seeing if
+// that object is in an index and enqueueing the value object from the indexer
+func processNextWatchObj(watchEvent watch.Event, queue workqueue.RateLimitingInterface, indexer cache.Indexer, index string) {
+	obj := watchEvent.Object
+	acc, err := meta.Accessor(obj)
+	if err != nil {
+		log.Errorf("Error processing object from watch: %v", err)
+		return
```

Comment:
Is there a reason you don't surface these errors to the caller? I'm afraid that we might get into an error hotloop which we will not back off from with the exponential backoff